### DISCUSSION TITLE
阻止垃圾代码提交

### DISCUSSION THREAD

**Discussion by maczyt** [maczyt/loonflow#4] [Ideas]
使用`lint-stage` 在 `git commit` 的时候进行 `eslint check`，如果有垃圾代码，那不允许`commit`

- `husky`
- `lint-stage`
- `eslint`

**Comment by maczyt** [comment]
后续对main分支lock，mr需要跑pipeline来检测

**Comment by blackholll** [comment]
后端有ut。actions里面可以看到历史执行记录。   block merge 这个我们后面研究下

**Reply by maczyt** [reply]
block merge 等上线了再研究

**Comment by maczyt** [comment]
how to add
``` shell
npx husky-init && npm install

# 修改pre-commit
npx lint-stage
```

> lint-stage

``` json
"lint-staged": {
    "*.ts(x)?": "eslint"
  }
```
